modules/transport/base/src/main/java/org/apache/axis2/transport/base/BaseConstants.java

```diff
@@ -131,5 +131,10 @@
     /** A message level property indicating a request to rollback the transaction associated with the message */
     public static final String SET_ROLLBACK_ONLY = "SET_ROLLBACK_ONLY";
     /** A message level property indicating a commit is required after the next immidiate send over a transport */
-    public static final String JTA_COMMIT_AFTER_SEND = "JTA_COMMIT_AFTER_SEND";    
+    public static final String JTA_COMMIT_AFTER_SEND = "JTA_COMMIT_AFTER_SEND";
+
+    /** The message context property name which holds the transaction is counted internally or not */
+    public static final String TRANSACTION = "internalTransaction";
```

Comment:
I was thinking about https://github.com/wso2/micro-integrator/pull/1537#discussion_r423663678 WDYT?

Reply:
Yeah, that will be better. Changed the property name as INTERNAL_TRANSACTION_COUNTED and true/false as its value.